framework/src/auxkernels/ProjectionAux.C

```diff
@@ -0,0 +1,95 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ProjectionAux.h"
+
+registerMooseObjectRenamed("MooseApp", SelfAux, "01/30/2024 24:00", ProjectionAux);
+registerMooseObject("MooseApp", ProjectionAux);
+
+InputParameters
+ProjectionAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription(
+      "Returns the specified variable as an auxiliary variable with a projection of the source "
+      "variable. If they are the same type, this amounts to a simple copy.");
+  params.addRequiredCoupledVar("v", "Variable to take the value of.");
+
+  // Technically possible to project from nodal to elemental and back
+  params.set<bool>("_allow_nodal_to_elemental_coupling") = true;
+
+  // We need some ghosting for all elemental to nodal projections
+  params.addParam<unsigned short>("ghost_layers", 1, "The number of layers of elements to ghost.");
+  params.addRelationshipManager(
+      "ElementPointNeighborLayers",
+      Moose::RelationshipManagerType::GEOMETRIC | Moose::RelationshipManagerType::ALGEBRAIC,
+      [](const InputParameters & obj_params, InputParameters & rm_params)
+      {
+        rm_params.set<unsigned short>("layers") = obj_params.get<unsigned short>("ghost_layers");
+        rm_params.set<bool>("use_displaced_mesh") = obj_params.get<bool>("use_displaced_mesh");
+      });
+  return params;
+}
+
+ProjectionAux::ProjectionAux(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _v(coupledValue("v")),
+    _source_variable(getVar("v", 0)),
+    _source_sys(_source_variable->sys())
+{
+  // Output some messages to user
+  if (isNodal() && _source_variable->isNodal() && _source_variable->order() < _var.order())
+    mooseInfo(
+        "Target nodal variable is higher order than source, using volume-weighted projection");
+  else if (_source_variable->order() < _var.order())
+    mooseInfo("Projection lowers order, please expect a loss of accuracy");
+  else if (_var.feType().family == SIDE_HIERARCHIC)
+    paramError("variable", "SIDE_HIERARCHIC is not supported");
+}
+
+Real
+ProjectionAux::computeValue()
+{
+  if (!isNodal() || (_source_variable->isNodal() && _source_variable->order() >= _var.order()))
+    return _v[_qp];
+  // Handle elemental variable projection into a nodal variable
+  // AND nodal low order -> nodal higher order
+  else
```

Comment:
The `else` cases made me think of nodal patch recovery. @hugary1995 am I wrong for thinking of that?

Reply:
Ya, both are least squares fitting essentially. My understanding is that auxkernel doesn't construct a larger patch of element -- it's just extrapolating into the nodes, whereas NPR does the fitting over all nodal neighbors.